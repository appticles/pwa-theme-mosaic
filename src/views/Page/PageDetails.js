import 'react-router-modal/css/react-router-modal.css';

import React from 'react';
import { Link } from 'react-router-dom';
import { Container, Image, Header, Button } from 'semantic-ui-react';
import Moment from 'react-moment';
import PropTypes from 'prop-types';

import { pagePropType } from '../SideMenu/reducer';
import './style.css';

const PageDetails = (props) => {
  function getImage(sourceImg) {
    let imageSource;
    if (sourceImg) {
      imageSource = sourceImg[0].source_url;
      return imageSource;
    }
    return imageSource;
  }

  const { page } = props;
  const { author } = page._embedded;
  const featuredMedia = page._embedded['wp:featuredmedia'];

  return (
    <Container className="page">
      <Header dividing>
        <Link to="/">
          <Button size="huge" icon="close" />
        </Link>
        <div dangerouslySetInnerHTML={{ __html: page.title.rendered }} />
      </Header>
      <Header.Subheader>
        &nbsp;{props.texts.TEXTS && props.texts.TEXTS.BY_AUTHOR}&nbsp;<b>{author[0].name}</b>,&nbsp;<Moment format="MMMM DD, YYYY">
          {page.date}
        </Moment>
      </Header.Subheader>
      {featuredMedia ? <Image src={getImage(featuredMedia)} /> : ''}
      <Container textAlign="justified">
        <div dangerouslySetInnerHTML={{ __html: page.content.rendered }} />
      </Container>
    </Container>
  );
};

PageDetails.defaultProps = {
  texts: {
    TEXTS: {
      BY_AUTHOR: 'by',
    },
  },
};
PageDetails.propTypes = {
  page: pagePropType.isRequired,
  texts: PropTypes.shape({
    TEXTS: PropTypes.shape({
      BY_AUTHOR: PropTypes.string,
    }),
  }),
};

export default PageDetails;
