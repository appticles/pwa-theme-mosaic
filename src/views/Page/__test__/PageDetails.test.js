import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import PageDetails from '../PageDetails';

describe('Page details', () => {
  it('renders page details', () => {
    const page = {
      id: 12,
      parent: 43,
      title: {
        rendered: 'Some title',
      },
      slug: 'some-title',
      content: {
        rendered: 'Some content for some page',
      },
      link: 'some-link',
      author: 32,
      featured_media: 43,
      _embedded: {
        author: [{ id: 32, name: 'Some author' }],
        'wp:featuredmedia': [
          {
            id: 43,
            source_url: 'some-image',
          },
        ],
      },
      comment_status: 'some-status',
      status: 'some-status',
    };
    const output = shallow(<PageDetails page={page} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
