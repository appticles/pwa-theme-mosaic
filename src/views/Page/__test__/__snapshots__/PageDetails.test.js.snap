// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page details renders page details 1`] = `
<Container
  className="page"
>
  <Header
    dividing={true}
  >
    <Link
      replace={false}
      to="/"
    >
      <Button
        as="button"
        icon="close"
        size="huge"
      />
    </Link>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "Some title",
        }
      }
    />
  </Header>
  <HeaderSubheader>
     
    by
     
    <b />
    , 
    <t
      ago={false}
      calendar={false}
      element="time"
      filter={[Function]}
      format="MMMM DD, YYYY"
      fromNow={false}
      interval={60000}
      onChange={[Function]}
      toNow={false}
      unix={false}
      utc={false}
    />
  </HeaderSubheader>
  <Image
    as="img"
    ui={true}
  />
  <Container
    textAlign="justified"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "Some content for some page",
        }
      }
    />
  </Container>
</Container>
`;
